src/calendar-list/index.tsx

```diff
@@ -290,11 +288,12 @@ class CalendarList extends Component<CalendarListProps, State> {
   };
 
   renderItem = ({item}: any) => {
-    const {calendarStyle, horizontal, calendarWidth, testID, ...others} = this.props;
-
+    const {horizontal, calendarStyle, calendarWidth, testID, markedDates, ...others} = this.props;
+    // NOTE: now only 'item' and 'markedDates' change for the 3 calendar (item.getTime) items
     return (
       <CalendarListItem
         {...others}
+        markedDates={item.getTime ? markedDates : undefined}
```

Comment:
We can improve more by not passing to months that don't have any marking dates.. 
it basically can reduce render of 2 more items, but it still something

Reply:
Ok. Check my last commit please